contrib/cli/great_expectations_contrib/package.py

```diff
@@ -110,22 +116,132 @@ def update_package_state(self) -> None:
-        # TODO: Write logic to assign values to attrs
-        # This is a black box for now
-        # for diagnostic in diagnostics:
-        #     pass
-        raise NotImplementedError
-
-    def _retrieve_package_expectations_diagnostics(self) -> List[Diagnostics]:
+    def _update_attrs_with_diagnostics(
+        self, diagnostics: List[ExpectationDiagnostics]
+    ) -> None:
+        self._update_from_package_info("package_info.yml")
+        self._update_expectations(diagnostics)
+        self._update_dependencies("requirements.txt")
+        self._update_contributors(diagnostics)
+
+    def _update_from_package_info(self, path: str) -> None:
+        if not os.path.exists(path):
+            logger.warning(f"Could not find package info file {path}")
+            return
+
+        with open(path) as f:
+            data: dict = yaml.load(f.read())
+
+        if not data:
+            logger.warning(f"{path} is empty so exiting early")
+            return
+
+        # Assign general attrs
+        general = data.get("general")
+        if general:
+            for attr in ("package_name", "icon", "description"):
+                self[attr] = general.get(attr)
+
+        # Assign code owners
+        self.code_owners = []
+        code_owners = data.get("code_owners")
+        if code_owners:
+            for owner in code_owners:
+                code_owner = GitHubUser(**owner)
+                self.code_owners.append(code_owner)
+
+        # Assign domain experts
+        self.domain_experts = []
+        domain_experts = data.get("domain_experts")
+        if domain_experts:
+            for expert in domain_experts:
+                domain_expert = DomainExpert(**expert)
+                self.domain_experts.append(domain_expert)
+
+    def _update_expectations(self, diagnostics: List[ExpectationDiagnostics]) -> None:
+        expectations = []
+        status = {maturity.name: 0 for maturity in Maturity}
+
+        for diagnostic in diagnostics:
+            expectation = RenderedExpectation(
+                name=diagnostic.description.snake_name,
+                tags=diagnostic.library_metadata.tags,
+                supported=[],
+                status=diagnostic.maturity_checklist,  # Should be converted to the proper type
+            )
+            expectations.append(expectation)
+
+            expectation_maturity = diagnostic.library_metadata.maturity
+            status[expectation_maturity] += 1
+
+        self.expectations = expectations
+        self.expectation_count = len(expectations)
+
+        # Enum is all caps but status attributes are lowercase
+        lowercase_status = {k.lower(): v for k, v in status.items()}
+        lowercase_status["total"] = sum(status.values())
+        self.status = PackageCompletenessStatus(**lowercase_status)
+
+        maturity = max(status, key=status.get)  # Get the key with the max value
+        self.maturity = Maturity[maturity]
```

Comment:
We didn't fully define the criteria for package maturity but I took the route of making the most prevalent maturity at the Expectation level the label for the package.

I'm happy to alter this if we'd like.